### PR TITLE
Allow null arguments

### PR DESCRIPTION
**Pull request by delkopiso**
The GraphQL spec now officially includes a Null Value. This is especially useful in a scenario where we need to differentiate between the absence of an argument and an argument that is being unset.

Example:
```
mutation {
  user_update(id: "1", address: null) {
    id
    address
  }
}
```

### CODE REVIEW

File: addon/types/argument-set.js [hunk 1]
```diff
@@ -9,7 +9,11 @@ export default class ArgumentSet extends Array {
 
   push(...args) {
     super.push(...args.filter((arg) => {
-      return Ember.typeOf(arg.value) !== 'null';
+      if (Ember.typeOf(arg) === 'object') {
```
Comment (tennisonchan) [inline addon/types/argument-set.js]: Do we need to check the type here?

Comment (delkopiso) [inline addon/types/argument-set.js]: Yes because `arg` will have a `value` property if it's an object

File: tests/unit/types/argument-set-test.js [hunk 2]
```diff
@@ -49,9 +56,19 @@ test("it can be made from a query", function(assert) {
   assert.equal(set[1].value, 10);
 });
 
-test("filters out null arguments", function(assert) {
+test("does not filter out null arguments from a query", function(assert) {
   let set = ArgumentSet.fromQuery({ status: null, limit: 10 });
```
Comment (tennisonchan) [inline tests/unit/types/argument-set-test.js]: Would be useful to have more falsy values in the test like empty string, number zero, etc...

**Comment by tennisonchan** [comment]
What if certain value cannot be `null`? how do GraphQL handle that?

**Comment by delkopiso** [comment]
The server throws it out. This is a client library, and static validation is currently supported.

File: tests/unit/types/argument-set-test.js (hunk 2 above)
Comment (delkopiso) [inline tests/unit/types/argument-set-test.js]: Those values are valid.

Comment (tennisonchan) [inline tests/unit/types/argument-set-test.js]: Yes, I know. It makes sure we will not filtered out in the future. It should be in this test though.
